packages/astro/src/integrations/hooks.ts

```diff
@@ -55,10 +56,42 @@ async function withTakingALongTimeMsg<T>({
 				JSON.stringify(hookName),
 			)}...`,
 		);
-	}, timeoutMs);
-	const result = await hookResult;
-	clearTimeout(timeout);
-	return result;
+	}, 3000);
+	try {
+		return await hookFn();
+	} catch (err) {
+		integrationLogger.error(
+			`An unhandled error occurred while running the ${bold(JSON.stringify(hookName))} hook`,
+		);
+		throw err;
+	} finally {
+		clearTimeout(timeout);
+	}
+}
+
+async function runHookInternal<THook extends keyof BaseIntegrationHooks>({
```

Comment:
Might be worth adding a comment on this function? Not super obvious what it does without reading through this file.

Reply:
```suggestion
// Executes the specified hook of the integration if present, and handles loggers 
async function runHookInternal<THook extends keyof BaseIntegrationHooks>({
```

Reply:
I’d use JSDoc so it shows up as context elsewhere in the editor but the comment LGTM!